src/components/error_boundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import type { ReactNode } from 'react';
+
+interface Props {
+  children: ReactNode;
+}
+
+interface State {
+  error?: Error;
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      error: undefined,
+      hasError: false,
+    };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error('ErrorBoundary caught an error:', error, errorInfo);
+  }
+
+  refreshPage() {
+    localStorage.clear();
+    window.location.reload();
+  }
```

Comment:
Very harsh!
`localStorage.clear()` - can remove what is needed. 
`window.location.reload()` - not required, child components are already destroyed, they will be remounted when the state changes.

```suggestion
  refreshPage() {
   this.setState({ hasError: false, error: undefined }); // or this.setState(initState);
  }
```